src/browser/process_tracker.js

```diff
@@ -238,36 +260,57 @@ ProcessTracker.prototype.launch = function(winIdentity, config, resolve) {
     }
 };
 
-ProcessTracker.prototype.monitor = function(winIdentity, pid, lifetime) {
-    if (this._processes[pid]) {
-        throw new Error(`Error monitoring external process, already monitoring pid: '${pid}'`);
+ProcessTracker.prototype.monitor = function(winIdentity, options) {
+    let {
+        pid: pidRequested,
+        uuid: uuidRequested,
+        lifetime,
+        monitor: monitorRequested,
+    } = options;
+
+    let pid = parseInt(pidRequested, 10);
+    let processEntry = this._processes[pid] || {};
+
+    if (isNaN(pid)) {
+        throw new Error(`Error monitoring external process, invalid pid value specified.`);
```

Comment:
With the `elipc_strategy.ts` this is sure to be caught right? Throwing on a call not initiated via an api call is not guaranteed to be handled gracefully.